source/common/network/address_impl.cc

```diff
@@ -34,7 +34,20 @@ Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss,
     RELEASE_ASSERT(ss_len == 0 || ss_len == sizeof(sockaddr_in6));
     const struct sockaddr_in6* sin6 = reinterpret_cast<const struct sockaddr_in6*>(&ss);
     ASSERT(AF_INET6 == sin6->sin6_family);
-    return std::make_shared<Address::Ipv6Instance>(*sin6, v6only);
+    if (!v6only && IN6_IS_ADDR_V4MAPPED(&sin6->sin6_addr)) {
+#ifndef s6_addr32
+#if defined(__APPLE__) && defined(__MACH__)
```

Comment:
We've left it as just `#ifdef __APPLE__` elsewhere. I don't think this is actually mach-specific in any event, as it was copied from BSD.

Reply:
Changed as requested, thanks for the review!